ui/sections/storage/inspectors/volume-dataset.reel/volume-dataset.js

```diff
@@ -77,12 +77,15 @@ exports.VolumeDataset = AbstractInspector.specialize(/** @lends VolumeDataset# *
         value:function() {
             var self = this;
             this.snapshotType = this._sectionService.VOLUME_SNAPSHOT_TYPE;
-            this._sectionService.listVolumeSnapshots().then(function(snapshots) {
-                self.snapshots = snapshots;
-            });
             this.vmwareDatasetType = this._sectionService.VMWARE_DATASET_TYPE;
-            this._sectionService.listVmwareDatasets().then(function(vmwareDatasets) {
+            return Promise.all([
+                this._sectionService.listVolumeSnapshots(),
+                this._sectionService.listVmwareDatasets(),
+                this.application.dataService.getNewInstanceForType(Model.ReplicationOptions)
```

Comment:
We try to go away from that design, the storage-section-service should have a method that return a new instance of ReplicationOptions. :)